test/commands/force/apex/log/list.test.ts

```diff
@@ -43,93 +49,100 @@ const rawLogResult = {
-    {
-      Id: '07L5tgg0005PGdTnEAL',
-      Application: 'Unknown',
-      DurationMilliseconds: 75,
-      Location: 'Unknown',
-      LogLength: 450,
-      LogUser: {
-        Name: 'Test User',
-        attributes: {},
-      },
-      Operation: 'API',
-      Request: 'API',
-      StartTime: '2020-10-13T05:39:43+0000',
-      Status: 'Assertion Failed',
-    },
-    {
-      Id: '07L5tgg0005PGdTnFPL',
-      Application: 'Unknown',
-      DurationMilliseconds: 75,
-      Location: 'Unknown',
-      LogLength: 450,
-      LogUser: {
-        Name: 'Test User2',
-        attributes: {},
-      },
-      Operation: 'API',
-      Request: 'API',
-      StartTime: '2020-10-13T05:39:43+0000',
-      Status: 'Successful',
-    },
-  ],
-  warnings: [],
-};
-
 const logRecords = [rawLogResult.result[0], rawLogResult.result[1]];
 
 describe('force:apex:log:list', () => {
-  before(() => {
-    process.exitCode = 0;
+  const config = new Config({ root: resolve(__dirname, '../../package.json') });
+  let sandbox: sinon.SinonSandbox;
+  let logStub: sinon.SinonStub;
+  let tableStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    sandbox = createSandbox();
+    logStub = sandbox.stub(SfCommand.prototype, 'log');
+    tableStub = sandbox.stub(SfCommand.prototype, 'table');
+    sandbox.stub(Org, 'create').resolves(Org.prototype);
   });
 
-  test
-    .withOrg({ username: 'test@username.com' }, true)
-    .stub(LogService.prototype, 'getLogRecords', () => logRecords)
-    .stdout()
-    .command(['force:apex:log:list', '-o', 'test@username.com'])
-    .it('runs default command with cleaned table output', (ctx) => {
-      expect(ctx.stdout).to.include(
-        'APPLICATION DURATION (MS) ID                  LOCATION SIZE (B) LOG USER   OPERATION REQUEST START TIME               STATUS'
-      );
-      expect(ctx.stdout).to.include(
-        ' Unknown     75            07L5tgg0005PGdTnEAL Unknown  450      Test User  API       API     2020-10-13T05:39:43+0000 Assertion Failed'
-      );
-      expect(ctx.stdout).to.include(
-        'Unknown     75            07L5tgg0005PGdTnFPL Unknown  450      Test User2 API       API     2020-10-13T05:39:43+0000 Successful'
-      );
-    });
+  afterEach(() => {
+    sandbox.restore();
+  });
 
-  test
-    .withOrg({ username: 'test@username.com' }, true)
-    .stub(LogService.prototype, 'getLogRecords', () => logRecords)
-    .stdout()
-    .command(['force:apex:log:list', '-o', 'test@username.com', '--json'])
-    .it('should return cleaned log records with json parameter specified', (ctx) => {
-      const expectedResult = JSON.stringify(cleanResult, null, 2);
-      expect(ctx.stdout).to.equal(`${expectedResult}\n`);
-    });
+  it('will list 0 logs', async () => {
+    sandbox.stub(LogService.prototype, 'getLogRecords').resolves([]);
+    const result = await new List([], config).run();
+    expect(result).to.deep.equal([]);
+    expect(logStub.firstCall.args[0]).to.equal('No debug logs found in org');
```

Comment:
this pattern (first call) has been pretty brittle (ex: new warnings or info emitted)

You might want to assert that `some` logStub call has an arg that includes the text you expect.